File: leetcode/problems/c/best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int min = prices[0], profit = 0, max = 0;

    for (int i = 1; i < pricesSize; i++) {
        if (prices[i] < min) 
            min = prices[i];
        profit = prices[i] - min;
        if (max < profit)
             max = profit;
    }
    return max;
}

File: leetcode/problems/rs/best-time-to-buy-and-sell-stock.rs
// by @codeAbinash
// Time : O(n)
// Space : O(1)

pub struct Solution;

impl Solution {
   pub fn max_profit(prices: Vec<i32>) -> i32 {
      let mut l = 0;
      let mut r = 1;
      let mut max_profit = 0;

      while r < prices.len() {
         if prices[l] < prices[r] {
            let profit = prices[r] - prices[l];
            max_profit = max_profit.max(profit);
         } else {
            l = r;
         }
         r += 1;
      }

      max_profit
   }
}
